hudi-common/src/main/java/org/apache/hudi/common/table/view/TableFileSystemView.java

```diff
@@ -166,6 +166,24 @@ Stream<FileSlice> getLatestFileSlicesBeforeOrOn(String partitionPath, String max
      */
     Stream<FileSlice> getLatestMergedFileSlicesBeforeOrOn(String partitionPath, String maxInstantTime);
 
+    /**
+     * Stream all "merged" file slices before on an instant time including files under inflight instants.
+     * If a filegroup has a pending compaction request,
+     * (1) if the base file from compaction is not present, the file slice before and after
+     * compaction request instant is merged and returned;
+     * (2) if the base file from compaction is present, and the compaction is inflight, only if the
```

Comment:
now I get the rational for the previous fix. 
So, essentially, if current client is actually the compaction, and is polling `getLatestMergedFileSlice including inflight` , we should return the base file created by inflght compaction and ignore entire previous file slice (prev base file + all logs from previous slice).